### PR TITLE
Feature/us3860 leader sends message

### PR DESCRIPTION
**Pull request by Doug-Shannon**
Leader Sends message

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs [hunk 1]
```diff
@@ -113,6 +116,35 @@ public int GetRsvpYesCount(int groupId, int eventId)
             return response[0]?.RsvpYesCount ?? 0;
         }
 
+        public List<MpGroup> GetAllGroupsLeadByParticipant(int participantId)
+        {
+            const string COLUMNS = "Group_Participants.group_participant_id, Group_Participants.participant_id,  Group_Participants.group_id, Group_Participants.group_role_id";
+            string search = $"Group_Participants.participant_id = {participantId}";
+
```
Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs]: I think we can get the groups directly through the rest service lets add a story for next sprint.

Comment (JoeKer1) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs]: Table: Group_Participants
Select: Group_ID_Table.*
filter: Group_Participants.participant_id = {participantId} and Group_Role_ID = 22 AND Group_ID_Table.Group_Type = 9

File: Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs [hunk 2]
```diff
@@ -108,6 +111,49 @@ public int GetRsvpYesCount(int groupId, int eventId)
...
+        public bool GetIsLeader(int participantId, int groupType = -1)
+        {
+            const string COLUMNS = "Group_Participants.group_role_id";
+            string search = $"Group_Participants.participant_id = {participantId} and Group_Role_ID = 22";
+
+            if (groupType != -1) {
+                search += $"AND Group_ID_Table.Group_Type_ID = {groupType}";
+            }
+
+            var mpGroupParticipants = _ministryPlatformRest.UsingAuthenticationToken(_apiUserService.GetToken()).Search<MpGroupParticipant>(MpRestEncode(search), MpRestEncode(COLUMNS));
```
Comment (Doug-Shannon) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs]: done

File: Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs [hunk 3]
```diff
@@ -108,6 +111,49 @@ public int GetRsvpYesCount(int groupId, int eventId)
...
+        {
+            const string COLUMNS = "Group_ID_Table.Group_Name, Group_Participants.group_participant_id, Group_Participants.participant_id,  Group_Participants.group_id, Group_Participants.group_role_id";
+            string search = $"Group_Participants.participant_id = {participantId} and Group_Role_ID = 22";
+
+            if (groupType != -1)
+            {
+                search += $"AND Group_ID_Table.Group_Type_ID = {groupType}";
+            }
+
+            var groupParticipantRecords = _ministryPlatformRest.UsingAuthenticationToken(_apiUserService.GetToken()).Search<MpGroupParticipant>(MpRestEncode(search), MpRestEncode(COLUMNS));
```
Comment (Doug-Shannon) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupParticipantRepository.cs]: done

**Pull request merged**